src/components/dateTimePicker/index.tsx

```diff
@@ -108,79 +108,66 @@ export interface DateTimePickerProps {
   testID?: string;
 }
 
-interface DateTimePickerState {
-  prevValue?: Date;
-  value?: Date;
-}
-
 type DateTimePickerPropsInternal = DateTimePickerProps & BaseComponentInjectedProps;
 
-class DateTimePicker extends Component<DateTimePickerPropsInternal, DateTimePickerState> {
-  static displayName = 'DateTimePicker';
-
-  static defaultProps = {
-    ...TextField.defaultProps,
-    mode: MODES.DATE
-  };
-
-  chosenDate?: Date;
-  expandable = React.createRef<ExpandableOverlayMethods>();
+function DateTimePicker(props: DateTimePickerPropsInternal) {
+  const {
+    renderInput,
+    editable,
+    mode = MODES.DATE,
+    dateFormat,
+    timeFormat,
+    dateFormatter,
+    timeFormatter,
+    minimumDate,
+    maximumDate,
+    locale,
+    is24Hour,
+    minuteInterval,
+    timeZoneOffsetInMinutes,
+    themeVariant,
+    onChange,
+    dialogProps,
+    headerStyle,
+    // @ts-expect-error
+    useCustomTheme,
```

Comment:
Why ignore 'useCustomTheme' and not add ThemeComponent to the types? i.e. `(props: DurationPickerProps & ThemeComponen)`

Reply:
It was actually ignored before, I didn't add it
And we don't have `ThemeComponent` type in public (only private). 

We'll eventually get rid of it, if you remember our conversation about deprecating the `useCustomTheme` prop all together.

Reply:
Actually you have the `useCustomTheme` prop on `asBaseComponent`...